src/lib/markdowngen.ml

```diff
@@ -2,20 +2,21 @@ open Rpc.Types
 open Idl
 open Codegen
 
-let pretty_xml str =
-  let ob = Buffer.create (2 * String.length str) in
-  let i = Xmlm.make_input (`String (0,str)) in
-  let o = Xmlm.make_output  ~nl:true ~indent:(Some 2) (`Buffer ob) in
-  let rec pull i o depth =
-    Xmlm.output o (Xmlm.peek i);
-    match Xmlm.input i with
-    | `El_start _ -> pull i o (depth + 1)
-    | `El_end -> if depth = 1 then () else pull i o (depth - 1)
-    | `Data _ -> pull i o depth
-    | `Dtd _ -> pull i o depth
+let get_description desc =
+  let escape =
+    let translate = function
+      (* Escape special XML (and HTML) chars that might become tags *)
+      | '<' -> Some "&lt;"
+      | '>' -> Some "&gt;"
+      | '&' -> Some "&amp;"
+      (* Escape some special markdown chars - these are not part of ocamldoc markup language *)
+      | '*' | '_' | '[' | ']' | '(' | ')' | '#' | '!' as c ->
+        Some ("\\" ^ (String.make 1 c))
+      | _ -> None
+    in
+    Internals.encode translate
   in
-  pull i o 0;
-  Buffer.contents ob
+  String.concat " " desc |> escape
```

Comment:
Markdown is meant to be readable from the CLI, I don't want to see html escapes there :-/
On the other hand I don't really have a good solution here

Reply:
I had to do this to avoid issues like `<random_text_not_html_tag>` and `[ocamldoc code block] (comment)` getting messed up in the generated HTML :/

Reply:
In the future, we could parse the ocamldoc format using Angstrom, and convert it into nice readable markdown format using backticks. (This is why it would be better to further split up the package, to avoid introducing more dependencies into the core lib)